internal/services/blueprints/client/client.go

```diff
@@ -4,29 +4,40 @@
 package client
 
 import (
-	"github.com/Azure/azure-sdk-for-go/services/preview/blueprint/mgmt/2018-11-01-preview/blueprint" // nolint: staticcheck
+	"github.com/hashicorp/go-azure-sdk/resource-manager/blueprints/2018-11-01-preview/assignment"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/blueprints/2018-11-01-preview/blueprint"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/blueprints/2018-11-01-preview/publishedblueprint"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/common"
 )
 
 type Client struct {
-	AssignmentsClient         *blueprint.AssignmentsClient
-	BlueprintsClient          *blueprint.BlueprintsClient
-	PublishedBlueprintsClient *blueprint.PublishedBlueprintsClient
+	AssignmentsClient         *assignment.AssignmentClient
+	BlueprintsClient          *blueprint.BlueprintClient
+	PublishedBlueprintsClient *publishedblueprint.PublishedBlueprintClient
 }
 
-func NewClient(o *common.ClientOptions) *Client {
-	assignmentsClient := blueprint.NewAssignmentsClientWithBaseURI(o.ResourceManagerEndpoint)
-	o.ConfigureClient(&assignmentsClient.Client, o.ResourceManagerAuthorizer)
+func NewClient(o *common.ClientOptions) (*Client, error) {
+	assignmentsClient, err := assignment.NewAssignmentClientWithBaseURI(o.Environment.ResourceManager)
+	if err != nil {
+		return nil, err
```

Comment:
can we name the client with this:

```suggestion
		return nil, fmt.Errorf("building Assignments client: %+v", err)
```